lib/core/src/persist/mod.rs

```diff
@@ -187,26 +194,117 @@ impl Persister {
             ),
         )?;
 
-        if let Some(destination) = destination {
-            // Only store the destination if there is no payment_details entry else
-            // the destination is overwritten by the tx script_pubkey
-            con.execute(
-                "INSERT INTO payment_details (
-                    tx_id,
-                    destination,
-                    description 
-                )
-                VALUES (?1, ?2, ?3)
-                ON CONFLICT (tx_id)
-                DO UPDATE SET description = COALESCE(?3, description) 
-            ",
-                (ptx.tx_id, destination, description),
+        let mut trigger_sync = false;
+        if let Some(ref payment_tx_details) = payment_tx_details {
+            // If the update comes from the wallet tx:
+            // - Skip updating the destination from the script_pubkey
+            // - Skip syncing the payment_tx_details
+            Self::insert_or_update_payment_details_inner(
+                &tx,
+                payment_tx_details,
+                from_wallet_tx_data,
             )?;
+            if !from_wallet_tx_data {
+                self.commit_outgoing(
+                    &tx,
+                    &payment_tx_details.tx_id,
+                    RecordType::PaymentDetails,
+                    None,
+                )?;
+                trigger_sync = true;
+            }
+        }
+
+        tx.commit()?;
+
+        if trigger_sync {
+            self.sync_trigger.try_send(())?;
         }
 
         Ok(())
     }
 
+    fn insert_or_update_payment_details_inner(
+        con: &Connection,
+        payment_tx_details: &PaymentTxDetails,
+        skip_destination_update: bool,
+    ) -> Result<()> {
+        let destination_update = skip_destination_update
+            .not()
+            .then_some("destination = excluded.destination,")
+            .unwrap_or_default();
+        con.execute(
+            &format!(
+                "INSERT INTO payment_details (
+                tx_id,
+                destination,
+                description,
+                lnurl_info_json
+            )
+            VALUES (?, ?, ?, ?)
+            ON CONFLICT (tx_id)
+            DO UPDATE SET
+                {destination_update}
```

Comment:
Why don't we use the same coalesce statement for the destination like we do for lnurl info?

Reply:
The destination is not optional in the struct, so can't be set back to null